index.js

```diff
@@ -55,20 +55,29 @@ module.exports = function(search, replacement, options) {
         callback(null, file);
       }
 
-      if (options && options.skipBinary) {
-        istextorbinary.isText(file.path, file.contents, function(err, result) {
-          if (err) {
-            return callback(err, file);
+      if (options) {
+        if (options.passFilename && typeof replacement === 'function') {
+          var userReplacement = replacement;
+          replacement = function (match) {
```

Comment:
Hmm, this won't work for regex replace. More than one argument is passed to the function, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter